src/obj_dir.ml

```diff
@@ -196,41 +196,21 @@ let make_lib ~dir ~has_private_modules lib_name =
 let make_external_no_private ~dir =
   External (External.make ~dir ~has_private_modules:false)
 
-let all_obj_dirs (type path) (t : path t) ~mode : path list =
-  match t with
-  | External e -> External.all_obj_dirs e ~mode
-  | Local e -> Local.all_obj_dirs e ~mode
-  | Local_as_path e -> Local.all_obj_dirs e ~mode |> List.map ~f:Path.build
-
-let public_cmi_dir (type path) (t : path t) : path =
-  match t with
-  | External e -> External.public_cmi_dir e
-  | Local e -> Local.public_cmi_dir e
-  | Local_as_path e -> Path.build (Local.public_cmi_dir e)
-
-let byte_dir (type path) (t : path t) : path =
-  match t with
-  | External e -> External.byte_dir e
-  | Local e -> Local.byte_dir e
-  | Local_as_path e -> Path.build (Local.byte_dir e)
-
-let native_dir (type path) (t : path t) : path =
-  match t with
-  | External e -> External.native_dir e
-  | Local e -> Local.native_dir e
-  | Local_as_path e -> Path.build (Local.native_dir e)
+let elim_build
```

Comment:
What's "_build" referring to in this name? Maybe call this `elim_to_path`? (or even, more simply, `get_path`)